src/ec/packages/ec-component-media-container/media-container.html.twig

```diff
@@ -0,0 +1,75 @@
+{% spaceless %}
+{#
+  Parameters:
+  - "alt" (string) (default: ''),
+  - "image" (string) (default: ''),
+  - "extra_classes" (string) (default: ''),
+  - "sources" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "type" (string) (default: ''),
+      },
+      ...
+    ],
+  - "tracks" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "kind" (string) (default: ''),
+        "srcLang" (string) (default: ''),
+        "label" (string) (default: ''),
+      },
+      ...
+    ],
+  - "description" (string) (default: ''),
+  - "extra_attributes" (array) (default: []): format: [
+      {
+        "name" (string) (default: ''),
+        "value" (string) (default: '')
+      },
+      ...
+    ]
+#}
+
+{# Internal properties #}
+
+{% set _css_class = 'ecl-media-container' %}
+{% set _image = image|default('') %}
+{% set _description = description|default('') %}
+{% set _alt = image|default('') %}
+{% set _extra_attributes = '' %}
+{% set _sources = sources|default({}) %}
+{% set _tracks = tracks|default({}) %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %} 
+
+{# Print the result #}
+
+<figure class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+{%- if _sources is not empty -%}
+  <video class="ecl-media-container__media" controls="controls" poster="{{ _image }}" alt="{{ _alt }}">
+  {%- for source in _sources -%}
+    <source type="{{ source.type }}" src="{{ source.src }}" />
+  {%- endfor -%}
+{%- if tracks is not empty -%}
+  {%- for track in _tracks -%}
+    <track kind="{{ track.kind }}" srcLang="{{ track.srcLang }}" label="{{ track.label }}" src="{{ track.src }}" />
```

Comment:
mmhh...i see, but in this case this property (srcLang) is coming from ECL, is not anything "defined" here, not sure it would be worth altering the parameters of the component in twig because of this.
What do you think?

Reply:
With my experience so far, all variables which do not follow snake case get reformatted before the twig template. And twig templates accept only snake case. And we need to adapt that before data is passed to the twig template, like this [example](https://github.com/ec-europa/ecl-twig/blob/master/src/ec/packages/ec-component-social-media-follow/demo/data.js#L11).